tf/utils.go

```diff
@@ -136,6 +138,136 @@ func NewPassReport(plan *tfjson.Plan) types.PassReport {
 	return out
 }
 
+func NewCoverageReportFromState(state *tfjson.State) (types.CoverageReport, error) {
+	out := types.CoverageReport{
+		Coverages: make(map[types.Resource]*coverage.Model, 0),
+	}
+	if state == nil || state.Values == nil || state.Values.RootModule == nil || state.Values.RootModule.Resources == nil {
+		return out, nil
+	}
+	for _, res := range state.Values.RootModule.Resources {
+		if !strings.HasPrefix(res.Address, "azapi_") {
+			continue
+		}
+
+		id := ""
+		if v, ok := res.AttributeValues["id"]; ok {
+			id = v.(string)
+		}
+		resourceType := ""
+		if v, ok := res.AttributeValues["type"]; ok {
+			resourceType = v.(string)
+		}
+
+		body := map[string]interface{}{}
+		if bodyRaw, ok := res.AttributeValues["body"].(string); ok {
```

Comment:
>if [](http://example.com/codeflow?start=2&length=2)

shall we define a new function like `getBody...` for this `if` branch to prevent multiple level of `if` ?

Reply:
Thanks, I have changed the code.